### PR TITLE
lightgbm: init at 2023-02-18

### PR DESCRIPTION
**Pull request by nviets**
Adding lightgbm with cli tool, r-package, and other features. WIP. Basic CLI package builds. Mostly based on the xgboost package build. To do:

- R package started but needs patching
- CUDA version builds to 100% but fails on ld errors
- HDFS Version
- Java wrapper
- Test on other systems
- Should this shared library be used as an input to the python package?

@jbedo, @teh, @costrouc

### CODE REVIEW

**Comment by nviets** [comment]
Adding lightgbm package with most major features enabled, including the R package and GPU support. To do list above is complete.

**Comment by happysalada** [comment]
hey, I'm happy to help you get that merged!
You have a typo in one place, but other than that it's good to go.
If you have time to fix the typo, I would be happy to merge!
Sorry it took so long for someone to review this!

File: pkgs/development/libraries/lightgbm/default.nix [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    '' + lib.optionalString javaWrapper ''
+      cp -r java $out
+      cp -r com $out
+      cp -r lightgbmlib.jar $out
+    '' + ''
+    '' + lib.optionalString javaWrapper ''
+      cp -r java $out
+      cp -r com $out
+      cp -r lightgbmlib.jar $out
+    '' + lib.optionals rLibrary ''
```
Comment (happysalada) [inline pkgs/development/libraries/lightgbm/default.nix:111]:
```suggestion
    '' + lib.optionalString rLibrary ''
```

**Comment by happysalada** [comment]
@nviets tagging you just to be sure you get this.

File: pkgs/development/libraries/lightgbm/default.nix (hunk 1 above)
Comment (nviets) [reply pkgs/development/libraries/lightgbm/default.nix:111]: Thanks for the help! Giving this a try now.

**Pull request merged**